pyquil/quilbase.py

```diff
@@ -413,6 +417,42 @@ def num_args(self) -> int:
         return int(np.log2(len(self.permutation)))
 
 
+class DefGateByPaulis(DefGate):
+    """
+    Records a gate definition as the exponentiation of a PauliSum.
+    """
+
+    def __init__(self,
+                 gate_name: str,
+                 parameters: List[Parameter],
+                 arguments: List[QubitDesignator],
+                 body: 'PauliSum'):
+        if not isinstance(gate_name, str):
+            raise TypeError("Gate name must be a string")
+
+        if gate_name in RESERVED_WORDS:
+            raise ValueError("Cannot use {} for a gate name since it's a reserved word".format(gate_name))
```

Comment:
f-string?

Reply:
OK; I was mimicking the style of the rest of this file.